generator/src/test/java/com/scottlogic/deg/generator/fieldspecs/FieldSpecTests.java

```diff
@@ -0,0 +1,614 @@
+            .withNumericRestrictions(new MockNumericRestrictions(numericRestrictionsEqual), FieldSpecSource.Empty)
+            .withStringRestrictions(new MockStringRestrictions(stringRestrictionsEqual), FieldSpecSource.Empty)
+            .withTypeRestrictions(new MockTypeRestrictions(typeRestrictionsEqual), FieldSpecSource.Empty)
+            .withNullRestrictions(new MockNullRestrictions(nullRestrictionsEqual), FieldSpecSource.Empty)
+            .withDateTimeRestrictions(new MockDateTimeRestrictions(dateTimeRestrictionsEqual), FieldSpecSource.Empty)
+            .withGranularityRestrictions(new MockGranularityRestrictions(granularityRestrictionsEqual), FieldSpecSource.Empty)
+            .withFormatRestrictions(new MockFormatRestrictions(formatRestrictionsEqual), FieldSpecSource.Empty)
+            .withMustContainRestriction(new MockMustContainRestriction(mustContainRestrictionsEqual));
+        FieldSpec b = FieldSpec.Empty
+            .withSetRestrictions(new MockSetRestrictions(setRestrictionsEqual), FieldSpecSource.Empty)
+            .withNumericRestrictions(new MockNumericRestrictions(numericRestrictionsEqual), FieldSpecSource.Empty)
+            .withStringRestrictions(new MockStringRestrictions(stringRestrictionsEqual), FieldSpecSource.Empty)
+            .withTypeRestrictions(new MockTypeRestrictions(typeRestrictionsEqual), FieldSpecSource.Empty)
+            .withNullRestrictions(new MockNullRestrictions(nullRestrictionsEqual), FieldSpecSource.Empty)
+            .withDateTimeRestrictions(new MockDateTimeRestrictions(dateTimeRestrictionsEqual), FieldSpecSource.Empty)
+            .withGranularityRestrictions(new MockGranularityRestrictions(granularityRestrictionsEqual), FieldSpecSource.Empty)
+            .withFormatRestrictions(new MockFormatRestrictions(formatRestrictionsEqual), FieldSpecSource.Empty)
+            .withMustContainRestriction(new MockMustContainRestriction(mustContainRestrictionsEqual));
+
+        Assert.assertThat(a, not(equalTo(b)));
+    }
+
+    private static Stream<Arguments> partiallyUnequalProvider(){
+        return Stream.of(
+            Arguments.of(false, true, true, true, true, true, true, true, true),
+            Arguments.of(true, false, true, true, true, true, true, true, true),
+            Arguments.of(true, true, false, true, true, true, true, true, true),
+            Arguments.of(true, true, true, false, true, true, true, true, true),
+            Arguments.of(true, true, true, true, false, true, true, true, true),
+            Arguments.of(true, true, true, true, true, false, true, true, true),
+            Arguments.of(true, true, true, true, true, true, false, true, true),
+            Arguments.of(true, true, true, true, true, true, true, false, true),
+            Arguments.of(true, true, true, true, true, true, true, true, false)
+        );
+    }
+
+    private class MockSetRestrictions extends SetRestrictions{
```

Comment:
All of these mock restrictions should be replaced with mockito mocks

Reply:
Unfortunately not, mockito doesn't support asserting/verifying `equals` or `hashCode`
https://github.com/mockito/mockito/wiki/FAQ